### PR TITLE
[feature] Allow webhooks to define filters on the start and/or end of the channel name

### PR DESCRIPTION
**Pull request by stayallive**
In our app we have a lot of traffic on channels. For our purposes only a few channels need to actually fire a webhook so our app can track it's state. Adding a filter on the channel name for the webhook allow us to cut down on the amount of "useless" webhook traffic.

I propose we add a `channel_starts_with` and `channel_ends_with` option to a webhook that has to match before a webhook is fired (if both are defined both need to match). I opted to keep this as simple as possible and not try to deal with regex or multiple matchers per webhook but if wanted we can always discuss on how to do that the best.

Let me know what you think!

### CODE REVIEW

**Comment by rennokki** [comment]
@stayallive This makes sense. I will merge it as soon as the Github actions runners are better, [they seem really drunk atm](https://downdetector.com/status/github/). I will re-run the tests when things are better.

**Comment by stayallive** [comment]
Sweet, and no worries, they indeed seem to be on the fritz.

File: src/app.ts [hunk 1]
```diff
@@ -21,6 +21,8 @@ export interface WebhookInterface {
     url?: string;
     lambda_function?: string;
     event_types: string[];
+    channel_starts_with?: string;
+    channel_ends_with?: string;
```
Comment (rennokki) [inline src/app.ts:25]: Would it be a better idea to have a separate section like `filter` or something? That would be easier to group things, I guess. 🤔

Comment (stayallive) [reply src/app.ts:25]: Fine with me. Simply adding a `filter` and nest these 2 under or would you change the options themself to some other name too?

Comment (rennokki) [reply src/app.ts]: I was thinking this is a good idea for more filters, not just the channel name that starts or ends with something. Making it useful in the future.

**Comment by rennokki** [comment]
I have changed to `channel_name_*` to make it more sense, just like in the prev discussion. Good job! 👍

**Pull request merged**